openxr/examples/opengl_xlib.rs

```diff
@@ -0,0 +1,275 @@
+#[macro_use]
+extern crate glium;
+
+use std::ptr;
+use openxr as xr;
+use x11::{xlib, glx};
+use std::ffi::{c_void, CString};
+use glium::texture::{DepthTexture2d, DepthFormat, UncompressedFloatFormat, MipmapsOption};
+
+const GL_RGBA8: u32 = 0x8058;
+
+pub struct Backend {
+    context: glx::GLXContext,
+    display: *mut xlib::Display,
+    visual: *mut xlib::XVisualInfo,
+    fb_config: glx::GLXFBConfig,
+    drawable: x11::xlib::Drawable,
+}
+
+impl Backend {
+    pub fn new() -> Self {
+        let mut fbcount = 0;
+        let visual_attribs = [0];
+
+        let context_attribs = [
+            glx::GLX_RGBA,
+            glx::GLX_DEPTH_SIZE,
+            24,
+            glx::GLX_DOUBLEBUFFER,
+            0,
+        ];
+        unsafe {
+            let display = xlib::XOpenDisplay(ptr::null());
+            let root = xlib::XDefaultRootWindow(display);
+            let visual = glx::glXChooseVisual(display, 0, context_attribs.as_ptr() as *mut _);
+            let context =
+                glx::glXCreateContext(display, visual, 0 as glx::GLXContext, true as xlib::Bool);
```

Comment:
In the interests of being a good example, `glXCreateContextAttribsARB` should be used to create an OpenGL core context for some reasonably modern OpenGL version (say, 4.0).